docs/book/src/topics/vm-identity.md

```diff
@@ -186,67 +83,73 @@ spec:
   template:
     spec:
       identity: UserAssigned
-      location: ${AZURE_LOCATION}
-      osDisk:
-        diskSizeGB: 128
-        managedDisk:
-          storageAccountType: Premium_LRS
-        osType: Linux
-      sshPublicKey: ${AZURE_SSH_PUBLIC_KEY_B64:=""}
       userAssignedIdentities:
       - providerID: ${USER_ASSIGNED_IDENTITY_PROVIDER_ID}
-      vmSize: ${AZURE_NODE_MACHINE_TYPE}
----
+      ...
 ```
 
 The CAPZ controller will look for `UserAssigned` value in `identity` field under `AzureMachineTemplate`, and assign the user identities listed in `userAssignedIdentities` to the virtual machine.
 
 * In Machine Pool
 
 ```yaml
-apiVersion: cluster.x-k8s.io/v1alpha4
-kind: MachinePool
-metadata:
-  name: ${CLUSTER_NAME}-mp-0
-  namespace: default
-spec:
-  clusterName: ${CLUSTER_NAME}
-  replicas: ${WORKER_MACHINE_COUNT}
-  template:
-    spec:
-      bootstrap:
-        configRef:
-          apiVersion: bootstrap.cluster.x-k8s.io/v1alpha4
-          kind: KubeadmConfig
-          name: ${CLUSTER_NAME}-mp-0
-      clusterName: ${CLUSTER_NAME}
-      infrastructureRef:
-        apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
-        kind: AzureMachinePool
-        name: ${CLUSTER_NAME}-mp-0
-      version: ${KUBERNETES_VERSION}
----
 apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
 kind: AzureMachinePool
 metadata:
   name: ${CLUSTER_NAME}-mp-0
   namespace: default
 spec:
   identity: UserAssigned
-  location: ${AZURE_LOCATION}
-  template:
-    osDisk:
-      diskSizeGB: 30
-      managedDisk:
-        storageAccountType: Premium_LRS
-      osType: Linux
-    sshPublicKey: ${AZURE_SSH_PUBLIC_KEY_B64:=""}
-    vmSize: ${AZURE_NODE_MACHINE_TYPE}
   userAssignedIdentities:
   - providerID: ${USER_ASSIGNED_IDENTITY_PROVIDER_ID}
----
+  ...
 ```
 
 The CAPZ controller will look for `UserAssigned` value in `identity` field under `AzureMachinePool`, and assign the user identities listed in `userAssignedIdentities` to the virtual machine scale set.
 
 Similar to system assigned identity, you can use the `user-assigned-identity`, and `machinepool-user-assigned-identity` flavors by setting the `{flavor}` in `clusterctl generate cluster --flavor {flavor}` to use user-assigned managed identity in machine deployment, and machine pool respectively.
+
+### Service Principal (not recommended)
+
```

Comment:
should we also deprecate this?

Reply:
hmm... I can't think of a scenario user assigned identity would be insufficient. I'm just hesitant since it's been around for a long time. Might be a good issue to open and get some community feedback about.

Reply:
it's also the current default for all flavors except the managed identity ones and we're not testing user-assigned in e2e so we definitely need to do more work before we can deprecate this. +1 to opening an issue

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1534